cmd/gateway_server/server/main.go

```diff
@@ -0,0 +1,190 @@
+package main
```

Comment:
I think you should switch this: move main.go to the parent dir (cmd/gateway_server) and move the proto & things into a subdir.